### PR TITLE
feat: Assignment 14 Asynchronous Programming Initial Commit

### PR DESCRIPTION
**Pull request by Subhasri-V-P**
**Background:**
To understand Asynchronous Programming using async and await.

**Implementation:**
Understanding and Implementing 
1. Async/Await
2. Task Parallel Library
3. Multi-Threading
4. Multi-Layered Async/Await Operations
5. Deadlock Conditions
6. ConfigureAwait
7. Async Void and Async Task with Exceptions

### CODE REVIEW

**Review by Shaalini-Subrammanian — CHANGES_REQUESTED** [review]

File: src/AsynchronousProgramming/AsyncAndAwait/Program.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿// <copyright file="Program.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+namespace AsyncAndAwait
+{
+    /// <summary>
+    /// Program class - To download and display content from website.
+    /// </summary>
+    public class Program
```
[the remark this answers is not in the document]
Comment (Subhasri-V-P) [reply src/AsynchronousProgramming/AsyncAndAwait/Program.cs:10]: Program class name refactored.

File: src/AsynchronousProgramming/TaskParallelLibraryImplementation/Program.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    /// </summary>
+    public class Program
+    {
+        /// <summary>
+        /// Main method - Entry point of the application and performs parallel loop execution.
+        /// </summary>
+        /// <param name="args">String array of arguments.</param>
+        public static void Main(string[] args)
+        {
+            var randomNumber = new Random();
```
[the remark this answers is not in the document]
Comment (Subhasri-V-P) [reply src/AsynchronousProgramming/TaskParallelLibraryImplementation/Program.cs:20]: Main function is refactored and updated.

File: src/AsynchronousProgramming/MultiThreading/Program.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        public static void Main(string[] args)
+        {
+            AppDomain.CurrentDomain.UnhandledException += new UnhandledExceptionEventHandler(HandleUnhandledException);
+            MathematicalCalculations objectToPerformCalculations = new ();
+            objectToPerformCalculations.AddIntegersToTheList();
+
+            int sum = 0;
+            Thread threadOne = new (new ThreadStart(() => sum = objectToPerformCalculations.SumOfArray()));
+            threadOne.Start();
+            threadOne.Join();
```
[the remark this answers is not in the document]
Comment (Subhasri-V-P) [reply src/AsynchronousProgramming/MultiThreading/Program.cs:25]: Suggested change was changed.

File: src/AsynchronousProgramming/DebuggingDeadLock/Program.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        /// <summary>
+        /// Main - Avoids deadlock and prints Hello World.
+        /// </summary>
+        /// <param name="args">tring array of arguments.</param>
+        /// <returns>returns null task object.</returns>
+        public static async Task Main(string[] args)
+        {
+            Program programObject = new ();
+            await programObject.DeadlockMethod();
```
[the remark this answers is not in the document]
Comment (Subhasri-V-P) [reply src/AsynchronousProgramming/DebuggingDeadLock/Program.cs:20]: Added at the end.

File: src/AsynchronousProgramming/DebuggingDeadLock/Program.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        public async Task<string> SomeAsyncOperation()
+        {
+            await Task.Delay(1000);
+            return "Hello, World!";
+        }
+
+        /// <summary>
+        /// DeadLockMethod - Gets the result and prints it in the console.
+        /// </summary>
+        /// <returns>Task object to stop execution until completion.</returns>
```
[the remark this answers is not in the document]
Comment (Subhasri-V-P) [reply src/AsynchronousProgramming/DebuggingDeadLock/Program.cs:36]: Documentation updated.

**Review by Shaalini-Subrammanian — APPROVED** [review]

**Pull request merged**